converters/tn_converter.py

```diff
@@ -0,0 +1,194 @@
+        write_file(os.path.join(self.out_dir, 'package.json'), manifest_str)
+
+    def download_frame_file(self, url_to_download, out_dir):
+        dw_filename = url_to_download.rsplit('/', 1)[1]
+
+        if not dw_filename.endswith('.txt'):
+            quiet_print(self.quiet, 'Skipping non-dokuwiki file {0}.'.format(dw_filename))
+            return
+
+        chapter_dir = dw_filename.split('-')[0]
+        md_filename = dw_filename.split('-')[1].replace('.txt', '.md')
+        save_as = os.path.join(out_dir, chapter_dir, md_filename)
+        if not self.overwrite and os.path.isfile(save_as):
+            quiet_print(self.quiet, 'Skipping {0}.'.format(dw_filename))
+            return
+
+        # for test cases
+        if self.git_repo.startswith('file://'):
+            url_to_download = os.path.join(self.git_repo, 'master', 'obs', 'notes', 'frames', dw_filename)
+
+        try:
+            quiet_print(self.quiet, 'Downloading {0}...'.format(url_to_download), end=' ')
+            text = get_url(url_to_download)
+        finally:
+            quiet_print(self.quiet, 'finished.')
+
+        quiet_print(self.quiet, 'Converting {0} to markdown...'.format(dw_filename), end=' ')
+
+        text = dokuwiki_to_markdown(text)
+        text = self.update_tw_links(text)
+        text = self.remove_nav_links(text)
+
+        # # remove tags
+        # text = self.tag_re.sub(r'', text)
+        #
+        # # remove squiggly tags
+        # text = self.squiggly_re.sub(r'', text)
+        #
+        # # remove extra blank lines
+        # text = self.extra_blanks_re.sub(r'\n\n', text)
+
+        quiet_print(self.quiet, 'finished.')
+
+        quiet_print(self.quiet, 'Saving {0}...'.format(save_as), end=' ')
+        write_file(save_as, text)
+        quiet_print(self.quiet, 'finished.')
+
+    def update_tw_links(self, text):
+        search_results = self.tw_link_re.search(text)
+        if not search_results:
+            return text
+        else:
+            return self.tw_link_re.sub(self.replace_tw_link, text)
+
+    def replace_tw_link(self, match):
+        parts = match.group(2).split('|', 1)
+        if len(parts) == 1:
+            return '[{0}](https://git.door43.org/Door43/en-tw/src/master/content/{1}/{0}.md)'.format(parts[0], match.group(1))
+        else:
+            return '[{0}](https://git.door43.org/Door43/en-tw/src/master/content/{1}/{2}.md)'.format(parts[1], match.group(1), parts[0])
+
+    def remove_nav_links(self, text):
+        return self.nav_links_re.sub(r'', text)
+
+    def clean_text(self, text):
+        """
+        Cleans up text from possible DokuWiki and HTML tag pollution.
+        """
+        if self.html_tag_re.search(text):
+            text = self.html_tag_re.sub('', text)
+        if self.link_tag_re.search(text):
+            text = self.link_tag_re.sub('', text)
+        return text
+
+    def get_json_dict(self, download_url):
```

Comment:
The functions `clean_text` and `get_json_dict` should be refactored and put into the `common.py` file because they are duplicated in `obs_converter.py`.